actions_helper/commands/deregister_task_definition.py

```diff
@@ -0,0 +1,71 @@
+
+
+def deregister_task_definition(
+    ecs_client: BaseClient,
+    cluster: str,
+    service: str,
+    production_task_definition_output: Optional[CreateTaskDefinitionOutput] = None,
+    local_task_definition_output: Optional[CreateTaskDefinitionOutput] = None,
+    preflight_task_definition_output: Optional[CreateTaskDefinitionOutput] = None,
+):
+    primary_deployment_definition_arn = [
+        deployment["taskDefinition"]
+        for deployment in ecs_client.describe_services(
+            cluster=cluster,
+            services=[service],
+        )["services"][0]["deployments"]
+        if deployment["status"] == "PRIMARY"
+    ][0]
+
+    click.echo(f"{primary_deployment_definition_arn=}")
+
+    production_task_definition_to_deregister = local_task_definition_to_deregister = (
+        preflight_task_definition_to_deregister
+    ) = None
+
+    fail_pipeline = True
+
+    if production_task_definition_output is not None:
+        if primary_deployment_definition_arn == production_task_definition_output.latest_task_definition_arn:
+            # Do not deregister task definition for initial deployment
+            if production_task_definition_output.previous_task_definition_arn:
+                production_task_definition_to_deregister = (
+                    production_task_definition_output.previous_task_definition_arn
+                )
+                local_task_definition_to_deregister = local_task_definition_output.previous_task_definition_arn
+                preflight_task_definition_to_deregister = preflight_task_definition_output.previous_task_definition_arn
+            fail_pipeline = False
+        else:
+            production_task_definition_to_deregister = production_task_definition_output.latest_task_definition_arn
+            local_task_definition_to_deregister = local_task_definition_output.latest_task_definition_arn
+            preflight_task_definition_to_deregister = preflight_task_definition_output.latest_task_definition_arn
+
+    if production_task_definition_output is not None and production_task_definition_to_deregister:
+        click.echo(f"{production_task_definition_to_deregister=}")
+        ecs_client.deregister_task_definition(
+            taskDefinition=production_task_definition_to_deregister,
+        )
+
+    if local_task_definition_output is not None and local_task_definition_to_deregister:
+        click.echo(f"{local_task_definition_to_deregister=}")
+        ecs_client.deregister_task_definition(
+            taskDefinition=local_task_definition_to_deregister,
+        )
+
+    if preflight_task_definition_output is not None and preflight_task_definition_to_deregister:
+        click.echo(f"{preflight_task_definition_to_deregister=}")
+        ecs_client.deregister_task_definition(
+            taskDefinition=preflight_task_definition_to_deregister,
+        )
+
+    if fail_pipeline:
+        set_error(message="Deployment failed; rollback triggered.")
```

Comment:
I understand, that you translated everything from the YAML as it was before, but perhaps this Python code can be simplified. With all the nested if statements it is hard to read and understand what will happen at the end.